OpenRA.Mods.Common/UtilityCommands/CheckRuntimeAssembliesCommand.cs

```diff
@@ -0,0 +1,58 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace OpenRA.Mods.Common.UtilityCommands
+{
+	public class CheckRuntimeAssembliesCommand : IUtilityCommand
+	{
+		string IUtilityCommand.Name { get { return "--check-runtime-assemblies"; } }
+
+		bool IUtilityCommand.ValidateArguments(string[] args)
+		{
+			return true;
+		}
+
+		[Desc("Generates an error if any assembly filenames in the current domain don't match the given whitelist",
+			"[WHITELIST]")]
+		void IUtilityCommand.Run(Utility utility, string[] args)
+		{
+			var whitelist = args
+				.Skip(1)
+				.Select(a => Path.GetFileName(a))
+				.ToArray();
+
+			// Load the renderer assembly so we can check its dependencies
+			Assembly.LoadFile(Platform.ResolvePath(Path.Combine(".", "OpenRA.Platforms.Default.dll")));
+
+			var missing = new List<string>();
+			foreach (var a in AppDomain.CurrentDomain.GetAssemblies())
+			{
+				var assemblyName = Path.GetFileName(a.Location);
+				if (!whitelist.Contains(assemblyName))
```

Comment:
If this command is invoked without a whitelist arg isn't it always going to flag the `OpenRA.Platforms.Default.dll` which we are loading by default?

Reply:
Yes, and this is the intended behaviour.

Reply:
Why?

Reply:
The purpose of the command is to generate errors for any assemblies used by the game which hasn't been explicitly specified by the caller (in this case, the CI) as being accounted for.  If the caller isn't accounting for the platform DLL then the game won't be able to run, and so the error is correct.